services/provider/server/server.go

```diff
@@ -422,6 +422,45 @@ func (s *OCSProviderServer) getExternalResources(ctx context.Context, consumerRe
 
 	}
 
+	cbpList := &rookCephv1.CephBlockPoolList{}
+	err = s.client.List(ctx, cbpList, client.InNamespace(s.namespace))
+	if err != nil {
+		return nil, fmt.Errorf("failed to list cephBlockPools in namespace. %v", err)
+	}
+	if len(cbpList.Items) > 0 {
+		blockPoolMapping := []csiopv1a1.BlockPoolIdPair{}
+		for i := range cbpList.Items {
+			cephBlockPool := cbpList.Items[i]
+			if cephBlockPool.GetAnnotations()[util.BlockPoolMirroringTargetIDAnnotation] != "" {
```

Comment:
There is no point in this check. See my comment about listing only the blockpools with the annotation